lib/compile.ml

```diff
@@ -184,48 +205,48 @@ let next colors st s pos =
   State.follow_transition st ~color:(Color_map.Table.get colors (String.unsafe_get s pos))
 ;;
 
-let rec loop info ~colors ~positions s ~pos ~last st0 st =
+let rec loop re ~colors ~positions s ~pos ~last st0 st =
   if pos < last
   then (
     let st' = next colors st s pos in
     let state_info = State.get_info st' in
     let idx = state_info.idx in
     if Idx.is_idx idx
     then (
-      Array.unsafe_set positions (Idx.idx idx) pos;
-      loop info ~colors ~positions s ~pos:(pos + 1) ~last st' st')
+      Positions.set positions (Idx.idx idx) pos;
+      loop re ~colors ~positions s ~pos:(pos + 1) ~last st' st')
     else if Idx.is_break idx
     then (
-      Array.unsafe_set positions (Idx.break_idx state_info.idx) pos;
+      Positions.set positions (Idx.break_idx state_info.idx) pos;
       st')
     else (
       (* Unknown *)
-      validate info s ~pos st0;
-      loop info ~colors ~positions:info.positions s ~pos ~last st0 st0))
+      validate re positions s ~pos st0;
+      loop re ~colors ~positions s ~pos ~last st0 st0))
   else st
 ;;
 
-let rec loop_no_mark info ~colors s ~pos ~last st0 st =
+let rec loop_no_mark re positions ~colors s ~pos ~last st0 st =
```

Comment:
One should not need any position here. One can pass a dummy value to `validate` or have a specific version of this function that does not expect positions.

Reply:
By the way, I think that having eight parameters is still ok, but beyond that, some parameters will be put on the stack on some architectures.

Reply:
> One should not need any position here. One can pass a dummy value to validate or have a specific version of this function that does not expect positions.

Done.

> By the way, I think that having eight parameters is still ok, but beyond that, some parameters will be put on the stack on some architectures.

Indeed. We're back to 7 now.